src/main/java/org/thethingsnetwork/java/app/lib/Message.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Shareif.com CONFIDENTIAL
+ * ________________________
+ *
+ * Copyright 2016 Shareif.com SPRL
+ * All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Shareif.com SPRL and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Shareif.com SPRL
+ * and its suppliers and may be covered by Belgian and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Shareif.com SPRL.
+ */
+package org.thethingsnetwork.java.app.lib;
+
+import java.util.Base64;
+import org.json.JSONException;
+import org.json.JSONObject;
+import static org.json.JSONObject.quote;
+
+/**
+ * This is a wrapper class for JsonObject to provide support for base64-encoded data
+ * @author Romain Cambier <romain@shareif.com>
+ */
+public class Message extends JSONObject {
+
+    public byte[] getBinary(String _key) {
+        Object object = get(_key);
+        if (object instanceof String) {
```

Comment:
Since the JsonObject can't handle a non-json type (bool, int, double, string, null, array, object), I have to decode it on-the-fly, That's why I can't use the same things you do in node.

I'll implement a simpler approach: a single getRawPayload() method so that it'll be as close as possible from your node implementation